src/DataWeaver.jl

```diff
@@ -1,5 +1,20 @@
+"""
+Module DataWeaver
+"""
+
 module DataWeaver
 
-# Write your package code here.
+export adios2_init, write_setup, write, read_setup, reading, read, finalize_adios,
+       inspect_variables
+
+let
+    global adios2_init, write_setup, write, read_setup, reading, reading_now, read,
```

Comment:
In order to have persistent internal variables, you do not need to make them `global`; you can simply initialize them at the beginning of the let block (see this example: https://github.com/eth-cscs/ImplicitGlobalGrid.jl/blob/master/src/update_halo.jl#L415-L441).
Making things global at the beginning of the let block, has the purpose of making things accessible outside of it - in fact it makes it globally accessible as the name suggests. This is absolutely fine for functions, but not for variables, because global variables lead to important performance penalties.